include/MiniLua/values.hpp

```diff
@@ -664,6 +664,23 @@ auto operator==(const UnaryOrigin&, const UnaryOrigin&) noexcept -> bool;
 auto operator!=(const UnaryOrigin&, const UnaryOrigin&) noexcept -> bool;
 auto operator<<(std::ostream&, const UnaryOrigin&) -> std::ostream&;
 
+/**
+ * Origin for a Value that was created in a nary operation (a function that receives n arguments)
+ * (or some functions with one argument) using val.
+ */
+struct NaryOrigin {
+    using ReverseFn = std::optional<SourceChangeTree>(const Value&, const Vallist&);
+
+    owning_ptr<Vallist> values;
```

Comment:
I think `owning_ptr` is not needed here (because Vallist is a complete type here).